tests/catch/unit/stream/hipStreamAttachMemAsync.cc

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright (c) 2022 Advanced Micro Devices, Inc. All rights reserved.
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+// TODO Enable it after hipStreamAttachMemAsync is feature complete on HIP
+
+#include <hip_test_common.hh>
+#include <memory>
+
+__device__ __managed__ int var = 0;
+
+enum class StreamAttachTestType { NullStream = 0, StreamPerThread, CreatedStream };
+
+TEST_CASE("Unit_hipStreamAttachMemAsync_Negative") {
+  hipStream_t stream{nullptr};
+
+  auto streamType =
+      GENERATE(StreamAttachTestType::NullStream, StreamAttachTestType::StreamPerThread,
+               StreamAttachTestType::CreatedStream);
+
+  if (streamType == StreamAttachTestType::StreamPerThread) {
+    stream = hipStreamPerThread;
+  } else if (streamType == StreamAttachTestType::CreatedStream) {
+    HIP_CHECK(hipStreamCreate(&stream));
+    REQUIRE(stream != nullptr);
+  }
+
+  SECTION("Invalid Resource Handle") {
+    int definitelyNotAManagedVariable = 0;
+    HIP_CHECK_ERROR(hipStreamAttachMemAsync(stream, &definitelyNotAManagedVariable, sizeof(int),
```

Comment:
dev_ptr needs to be type void* and not int*. Causes compilation failures. Similar issue is seen in the rest of the file as well.

Reply:
done